tests/jballerina-unit-test/src/test/resources/test-src/statements/variabledef/record-variable-definition-stmt.bal

```diff
@@ -468,10 +468,145 @@ function testRestFieldResolving() {
     testClosedRecordDefinedRestField();
 }
 
+type Topt1 record {
+    int x?;
+    int y?;
+};
+
+function testOptionalFieldAssignment1() {
+    Topt1 topt = {x: 2, y: 4};
+    var {x, y} = topt;
+    assertEquality(2, x);
+    assertEquality(4, y);
+
+    topt = {y: 5};
+    var {x:x1, y:y1} = topt;
+    assertTrue(x1 is ());
+    assertEquality(5, y1);
+
+    topt = {};
+    var {x:x2, y:y2} = topt;
+    assertTrue(x2 is ());
+    assertTrue(y2 is ());
+}
+
+type Topt2 record {
+    int a;
+    record {
+        int b?;
+    }[1] c;
+};
+
+function testOptionalFieldAssignment2() {
+    Topt2 topt = {a: 4, c: [{b: 5}]};
+    var {a, c: [{b}]} = topt;
+    assertEquality(4, a);
+    assertEquality(5, b);
+}
+
+function testOptionalFieldAssignment3() {
+    Topt2 topt = {a: 4, c: [{b: 5}]};
+    var {a, c: [{}]} = topt;
+    assertEquality(4, a);
+}
+
+type Topt3 record {
+    int a;
+    record {
+        int b?;
+    }[1] c?;
+};
+
+function testOptionalFieldAssignment4() {
+    Topt3 topt = {a: 4, c: [{b: 5}]};
+    var {a, c} = topt;
+    assertEquality(a, 4);
+    record {int b?;}[1]? _ = c;
+    var {b} = (<record {int b?;}[1]> c)[0];
+    assertEquality(b, 5);
+}
+
+function testOptionalFieldAssignment5() {
+    Topt3 topt = {a: 4};
+    var {a, c} = topt;
+    assertTrue(c is ());
+}
+
+function testOptionalFieldAssignment6() {
+    [record {int i?;}] y = [{}];
+    var [{i}] = y;
+    assertTrue(i is ());
+    var [{i: i2}] = y;
+    assertTrue(i2 is ());
+}
+
+function testRecordDefinitionWithOptionalFields1() {
+    int? x = 2;
+    int? y = ();
+    Topt1 topt = {x, y};
+    assertEquality(topt.x, 2);
+    assertTrue(topt.y is ());
+}
+
+function testRecordDefinitionWithOptionalFields2() {
+    int? x = ();
+    int? y = ();
+    Topt1 topt = {x, y};
+    assertTrue(topt.x is ());
+    assertTrue(topt.y is ());
+}
+
+function testRecordDefinitionWithOptionalFields3() {
+    Topt1 topt = {x: 3, y: ()};
+    assertEquality(topt.x, 3);
+    assertTrue(topt.y is ());
+}
+
+function testRecordDefinitionWithOptionalFields4() {
+    int? a = 4;
+    int? b = ();
+    Topt2 topt = {a: 4, c: [{b}]};
+    assertEquality(4, topt.a);
+    assertTrue((topt.c)[0].b is ());
+}
+
+function testRecordDefinitionWithOptionalFields5() {
+    int? a = 4;
+    record {
+        int b?;
+    }[1]? c = ();
+    Topt3 topt = {a: 4, c};
+    assertEquality(4, topt.a);
+    assertTrue(topt.c is ());
+}
+
+function testRecordDefinitionWithOptionalFields6() {
+    int? a = 4;
+    int? b = 41;
+    record {
+        int b?;
+    }[1]? c = ();
+    Topt3 topt = {a: 4, c: [{b}]};
+    assertEquality(4, topt.a);
+    assertTrue(topt.c is ());
+}
+
+function testRecordDefinitionWithOptionalFields7() {
+    int x = 2;
+    int? y = ();
+    Topt1 topt = {x, y};
+    assertEquality(topt.x, 2);
+    assertTrue(topt.y is ());
+}
+
```

Comment:
Shall we add some tests to check whether the query cases now works same for optional fields and nillable types as mentioned in https://github.com/ballerina-platform/ballerina-spec/issues/1079?

Reply:
Added